llvm/lib/CodeGen/MachineOperand.cpp

```diff
@@ -1240,7 +1240,7 @@ void MachineMemOperand::print(raw_ostream &OS, ModuleSlotTracker &MST,
        << "unknown-address";
   }
   MachineOperand::printOperandOffset(OS, getOffset());
-  if (getSize() > 0 && getAlign() != getSize())
+  if (getType().getElementCount().getKnownMinValue() > 0 && getAlign() != getType().getElementCount().getKnownMinValue())
```

Comment:
```suggestion
unsigned MinSize = getType().getSizeInBytes().getKnownMinValue();
if (MinSize  > 0 && getAlign() != MinSize)
```
I think this should solve a large chunk of the test failures. The old code is saying `if (the size of the memory reference is not equal to the alignment)`. 

The code you have here is saying `if (the minimum number of elements is not equal to the alignment)`.

However, we probably need to account for the size of the element as well. The tricky part here is that we only know the minimum size of the memory reference. In the case that the MinSize is not equal to the alignment, it can't hurt to emit an explicit align value.

Reply:
`getSize()` checks `isValid()` so it probably makes sense to change my suggestion to 
```suggestion

unsigned MinSize = 
  MemoryType.isValid() ? getType().getSizeInBytes().getKnownMinValue() : 0;
if (MinSize  > 0 && getAlign() != MinSize)
```

Reply:
And maybe to leave a TODO for `getSize()` to return a `TypeSize`?

Reply:
yep!

Reply:
> `getSize()` checks `isValid()` so it probably makes sense to change my suggestion to

Yes totally! Actuall I was just trying to push up to use GitHub actions to check because for some reason when I tried to test the [test_g_load.mir](https://github.com/llvm/llvm-project/blob/main/llvm/test/MachineVerifier/test_g_load.mir) file locally, I got errors like:
```C
*** Bad machine code: atomic load cannot use release ordering ***
- function:    test_load
- basic block: %bb.0  (0x560485132100)
- instruction: %5:_(s32) = G_LOAD %2:_(p0) :: (load acq_rel (s32))
LLVM ERROR: Found 5 machine code errors.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
```
But GitHub action is no longer reporting that so I'm a bit confused.

The command I used in the terminal is to first change my build to: `-DLLVM_TARGETS_TO_BUILD="RISCV;AArch64;ARM"; and then I used `./build-llc/bin/llc  -mtriple=arm64 -run-pass=none -verify-machineinstrs llvm/test/MachineVerifier/test_g_load.mir` to test this file.

Reply:
Does it fail when you run `ninja check-llvm`?

Reply:
If you look at the CHECK statements, this test is expecting bad machine code failures.

Reply:
it can be useful to do `./build/bin/llvm-lit -v llvm/test/MachineVerifier/test_g_load.mir` to see whether a specific test is failing/passing and why.

Reply:
> it can be useful to do ./build/bin/llvm-lit -v llvm/test/MachineVerifier/test_g_load.mir to see whether a specific test is failing/passing and why.

this command is passing!

> Does it fail when you run ninja check-llvm?

I tried to run it but every time I use `check-llvm` it puts a lot of burden and my machine just got stuck there or even freezed.

> If you look at the CHECK statements, this test is expecting bad machine code failures.

right, and I thought there would be a try-catch mechanism to catch the expected failures to report "there are expected failures" instead of actually raising them. 

But anyhow, `llvm-lit` is working, thanks!